cppapi/server/jpeg_mmx/jpeg_dct_mmx.cpp

```diff
@@ -253,7 +253,7 @@ ALIGN8 short __jpmm_row_tab_frw[] = {  // forward_dct coeff table
 
 #if __GNUC__ > 3
 #define lea_addr_in_reg(REG1) \
-	"call	__i686.get_pc_thunk.bx	\n" \
+	"call	__x86.get_pc_thunk.bx	\n" \
```

Comment:
When looking at the commit @t-b referenced in his commit message:
https://github.com/gcc-mirror/gcc/commit/a3330c9d558720697db4f6f767252335eefe89a3
it seems like this commit is part of gcc releases >= 4.7.0.
So what happens for 4.0.0 <= gcc versions < 4.7.0?
This is not so important for tango-9-lts since it requires recent gcc versions for the C++11 features, but this info could be useful if we backport the fix into 9.3-backports.
For tango-9-lts, we could also remove at some point the #ifdef sections when __GNUC__ <= 4 I think.

Reply:
Valid point. How about removing the `__GNUC__ > 3` branch and always use the fallback? The fallback code uses the exact same code as the GCC code.

See
https://github.com/tango-controls/cppTango/blob/8d0bd12e18aa7ea5dad4642790e68499705623b7/cppapi/server/jpeg_mmx/jpeg_dct_mmx.cpp#L1974-L1990

vs

 https://github.com/gcc-mirror/gcc/commit/a3330c9d558720697db4f6f767252335eefe89a3#diff-cb9d1aa158c2bbb17eb20b4b216cfd3f93dd41d65332c9d70991a829b14aaccaL463

Reply:
@mliszcz @bourtemb I went that route and removed the GCC specific code.